lib/pbench/server/cache_manager.py

```diff
@@ -646,6 +665,41 @@ def extract(tarball_path: Path, path: str) -> Inventory:
             tarball_path, f"Unexpected error from {tar_path}: {error_text!r}"
         )
 
+    def stream(self, path: str) -> Inventory:
+        """Return a cached inventory file as a binary stream
+
+        Args:
+            path: Relative path of a regular file within a tarball.
+
+        Returns:
+            An Inventory object encapsulating the file stream and the cache
+            lock. The caller is reponsible for closing the Inventory!
+        """
+        self.cache_create()
+
+        # NOTE: the cache is unlocked here. That's "probably OK" as the reclaim
+        # is based on the last modified timestamp of the lock file and only
+        # runs periodically.
+        #
+        # TODO: Can we convert the lock from EXCLUSIVE to SHARED? If so, that
+        # would be ideal.
```

Comment:
I don't think this is super-important, unless we do a manual cache flush that happens to hit right here. (And, yeah, I thought in the middle of the night that I should add a fail-out if `self.unpacked` goes away once we re-lock, but forgot. I should probably still add it, but it's a minor issue since under normal circumstances it'd require that the `gunicorn` process hibernates for 4 hours here. So, yeah, I should do it; but I really don't want to fail, which means I need to catch it and loop back to unpack again, and that's ugly.)